documentation/openapi.yaml

```diff
@@ -0,0 +1,121 @@
+openapi: 3.0.0
+info:
+  version: 1.0.0-oas3
+  title: Kubportal Backend API
+paths:
+  /statistics/{stat}:
+    get:
+      parameters:
+        - in: path
+          name: stat
+          required: true
+          schema:
+            type: string
+          description: The statistic type being fetched
+      responses:
+        '200':
+          description: A single statistic value
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  value:
+                    type: string
+                    example: 1.0.0
+  /webapps/:
+    get:
+      responses:
+        '200':
+          description: A list of web applications allowed for the user
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: object
+                  properties:
+                    link_name:
+                      type: string
+                      example: kuchen
+                    link_url:
+                      type: string
+                      example: 'https://mango-saft.com'
+                      
+  /users/{uid}:
+    get:
+      parameters:
+        - in: path
+          name: uid
+          required: true
+          schema:
+            type: integer
+          description: An user object 
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                properties:
+                  id:
+                    type: integer
+                    example: 1
+                  firstname: 
+                    type: string
+                    example: 'Mandarin'
+                  name:
+                    type: string
+                    example: 'Orange'
+                  username:
+                    type: string
+                    example: 's77777'
+                  portal_groups:
+                    type: array
+                    items:
+                      type: integer
+                    example: ['1', '2']
+                  service_account:
+                    type: string
+                    example: 'default'
+                  kubernetes_namespace:
+                    type: string
+                    example: 'default'
+
+                    
+  /login:
+    post:
+      summary: Authorize an user
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                type: object
+                properties:
+                  username:
+                    type: string
+                    example: 's77777'
+                  password:
+                    type: string
+                    example: 'test'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
```

Comment:
I checked the result of the current backend implementation, and it looks like this:

`{
    "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTk3OTE4MjkwLCJqdGkiOiIxYmYzOTEyMjUyODc0ZmE1YTdmYTc1NDc3Y2RjZDMwOSIsInVzZXJfaWQiOjF9.9a6A3_el-ue_bXCxYecXrw6yXR_fcvCquaBmaCLaJCk",
    "refresh_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoicmVmcmVzaCIsImV4cCI6MTU5ODAwNDM5MCwianRpIjoiMjhlNmViNDE3ZGFkNGQ4MDlhMGQzMzJkZTAxNjkyZmIiLCJ1c2VyX2lkIjoxfQ.VtXbLO-eQmfH3n_RBjwwecn0CMhCFFLXJ5JJPYI1v1w",
    "user": {
        "pk": 1,
        "username": "root",
        "email": "",
        "first_name": "",
        "last_name": ""
    }
}
`

It would be perfect to copy that data scheme here.

Besides that, this answer data structure might be a perfect candidate for a "components" section in the OpenAPI spec. The reason is that we have another API call (for the Google login) that returns exactly the same.

Please check this: https://swagger.io/docs/specification/components/